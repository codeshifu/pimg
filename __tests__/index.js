import React from 'react'
import { configure, mount, shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import EnzymeToJSON from 'enzyme-to-json'

// Setup Enzyme Adapter
configure({ adapter: new Adapter() })

import Image, { Thumbnail } from '../src/'

const image =
	'https://res.cloudinary.com/stackpie/image/upload/v1513979515/-895520106_m1whb3.jpg'

const thumbnail = image.replace('/upload/', '/upload/c_thumb,w_30/')

describe('pimg', () => {
	test('renders correctly', () => {
		const component = mount(<Image src={image} />)

		expect(EnzymeToJSON(component)).toMatchSnapshot()
	})
})

describe('Image', () => {
	test('generates thumbnail when loading', () => {
		const component = mount(<Image src={image} />)

		if (component.state().loading) {
			expect(component.state().thumbnail).toBe(thumbnail)
		}
	})

	test('can take a Thumnail child component', () => {
		const component = mount(
			<Image src={image}>
				<Thumbnail src={thumbnail} />
			</Image>
		)

		expect(EnzymeToJSON(component)).toMatchSnapshot()
	})

	test('can take a thumbnail prop', () => {
		const component = mount(
			<Image src={image} thumbnail='http://myThumb' />
		)

		expect(EnzymeToJSON(component)).toMatchSnapshot()
	})

	test('uses the default classNames when loading', () => {
		const component = shallow(<Image src={image} />)

		if (component.state().loading) {
			expect(component.props().className).toBe('pimg pimg__loading')
		}
	})

	test('uses the loadingClassName passed when loading', () => {
		const component = shallow(<Image className="myImage" loadingClassName="image_is_loading" src={image} />)

		if (component.state().loading) {
			expect(component.props().className).toBe('myImage image_is_loading')
		}
	})

	test('appends `__loading` if loadingClassName isn\'t passed', () => {
		const component = shallow(<Image className="myImage" src={image} />)

		if (component.state().loading) {
			expect(component.props().className).toBe('myImage myImage__loading')
		}
	})
})

describe('Thumbnail', () => {
	test('uses the thumbnail passed', () => {
		const thumbnail = 'xyz'

		const component = shallow(
			<Image src={image}>
				<Thumbnail src={thumbnail} />
			</Image>
		)

		if (component.state().loading) {
			expect(component.props().src).toBe(thumbnail)
		}
	})

	test('uses the className passed when loading', () => {
		const component = shallow(
			<Image className="myImage" src={image}>
				<Thumbnail className="myThumbnail" src={thumbnail} />
			</Image>
		)

		if (component.state().loading) {
			expect(component.props().className).toBe('myThumbnail')
			expect(component.props().imageClassName).toBe('myImage')
		}
	})
})
