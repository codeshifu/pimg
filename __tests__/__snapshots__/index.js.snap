// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image can take a Thumnail child component 1`] = `
<Image
  src="https://res.cloudinary.com/stackpie/image/upload/v1513979515/-895520106_m1whb3.jpg"
>
  <Thumbnail
    blob={null}
    fetchImage={[Function]}
    image="https://res.cloudinary.com/stackpie/image/upload/v1513979515/-895520106_m1whb3.jpg"
    loading={true}
    setThumbnail={[Function]}
    src="https://res.cloudinary.com/stackpie/image/upload/c_thumb,w_30/v1513979515/-895520106_m1whb3.jpg"
  >
    <img
      className="pimg pimg_loading"
    />
  </Thumbnail>
</Image>
`;

exports[`Image can take a thumbnail prop 1`] = `
<Image
  src="https://res.cloudinary.com/stackpie/image/upload/v1513979515/-895520106_m1whb3.jpg"
  thumbnail="http://myThumb"
>
  <img
    className="pimg pimg__loading"
    src="http://myThumb"
  />
</Image>
`;

exports[`pimg renders correctly 1`] = `
<Image
  src="https://res.cloudinary.com/stackpie/image/upload/v1513979515/-895520106_m1whb3.jpg"
>
  <img
    className="pimg pimg__loading"
    src="https://res.cloudinary.com/stackpie/image/upload/c_thumb,w_30/v1513979515/-895520106_m1whb3.jpg"
  />
</Image>
`;
